src/audio/detect_test.c

```diff
@@ -142,26 +142,34 @@ static void default_detect_test(struct comp_dev *dev,
 				struct comp_buffer *source, uint32_t frames)
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
-
-	int16_t *src;
-	int16_t diff;
+	void *src;
+	int32_t diff;
 	uint32_t count = frames; /**< Assuming single channel */
 	uint32_t sample;
+	uint16_t valid_bits = dev->params.sample_valid_bytes * 8;
+	int32_t activation_threshold = cd->config.activation_threshold;
```

Comment:
I see it's helper read-only val. I'd make it const if checkpatch allows.

Reply:
Can be done.